source-sftp/.snapshots/TestSpec

```diff
@@ -1,33 +1,78 @@
 {
-  "$schema": "http://json-schema.org/draft/2020-12/schema",
-  "$id": "https://github.com/estuary/connectors/source-sftp/config",
+  "$schema": "http://json-schema.org/draft-07/schema#",
   "properties": {
     "address": {
       "type": "string",
       "title": "Address",
       "description": "Host and port of the SFTP server. Example: myserver.com:22",
       "order": 0
     },
-    "username": {
-      "type": "string",
-      "title": "Username",
-      "description": "Username for authentication.",
-      "order": 1
-    },
-    "password": {
-      "type": "string",
-      "title": "Password",
-      "description": "Password for authentication. Only one of Password or SSHKey must be provided.",
-      "order": 2,
-      "secret": true
-    },
-    "sshKey": {
-      "type": "string",
-      "title": "SSH Key",
-      "description": "SSH Key for authentication. Only one of Password or SSHKey must be provided.",
-      "multiline": true,
-      "order": 3,
-      "secret": true
+    "credentials": {
+      "type": "object",
+      "discriminator": {
+        "propertyName": "type"
+      },
+      "oneOf": [
+        {
+          "title": "SSH Key",
+          "properties": {
+            "type": {
+              "type": "string",
+              "title": "Authentication Method",
+              "const": "sshKey",
+              "default": "sshKey"
+            },
+            "username": {
+              "type": "string",
+              "title": "Username",
+              "description": "Username for authentication.",
+              "order": 1
+            },
+            "sshKey": {
+              "type": "string",
+              "title": "SSH Key",
+              "description": "SSH Key for authentication",
+              "multiline": true,
+              "order": 3,
```

Comment:
This order should be 2?

Reply:
Type is order 1, username is order 2, sshKey is order 3?

Reply:
Oh I must have been missing the ordering for the type!